inc/Engine/CriticalPath/RESTCSSSubscriber.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace WP_Rocket\Engine\CriticalPath;
+
+use WP_Rocket\Event_Management\Subscriber_Interface;
+
+/**
+ * Class RESTCSSSubscriber
+ *
+ * @package WP_Rocket\Engine\CriticalPath
+ */
+class RESTCSSSubscriber implements Subscriber_Interface {
+
+	/**
+	 * REST Post manager that has generate and delete methods.
+	 *
+	 * @var RESTWPPost
+	 */
+	private $rest_post_manager;
+
+	/**
+	 * RESTCSSSubscriber constructor.
+	 *
+	 * @param RESTWPPost $rest_post_manager Post manager instance.
+	 */
+	public function __construct( RESTWPPost $rest_post_manager ) {
+		$this->rest_post_manager = $rest_post_manager;
+	}
```

Comment:
This implementation is too restrictive as it's binds only the `RESTWPPost` object to this subscriber. Instead, let's use an interface and change the naming here:

```suggestion
	/**
	 * RESTCSSSubscriber constructor.
	 *
	 * @param RESTWPInterface $rest_manager Post manager instance.
	 */
	public function __construct( RESTWPInterface $rest_manager ) {
		$this->rest_manager = $rest_manager;
	}
```